doc/developer/platform/ux.md

```diff
@@ -285,31 +285,152 @@ The HTTPS server exposes two APIs:
 
   * `POST /query` executes a batch of SQL statements.
 
-    The request body specifies the SQL string to execute. The request may contain
-    multiple SQL statements separated by semicolons. The statements are submitted
-    for processing as if they were supplied via the [Simple Query][simple-query]
-    flow of the PostgreSQL protocol (i.e., "the statements are executed as a
-    single transaction unless explicit transaction control commands are included
-    to force a different behavior").
-
-    The response is a JSON array containing the results of each query in the
-    request.
+    The request body is JSON containing a key, `sql`, which specifies the SQL
+    string to execute.
+    - The request may contain multiple SQL statements separated by semicolons.
+    - The statements are submitted for processing as if they were supplied via
+    the [Simple Query][simple-query] flow of the PostgreSQL protocol (i.e., "the
+    statements are executed as a single transaction unless explicit transaction
+    control commands are included to force a different behavior").
+    - We abort the request upon encountering the first error.
+
+    The response is a JSON array containing the in-order results of each
+    executed transaction in the request, flattened so as to not structurally
+    delineate transactions (though understanding explicit transaction boundaries
+    is inferrable).
+
+    For instance if a user submitted data of:
+
+    ```json
+    '{
+      "sql":
+        "INSERT INTO mytable VALUES(1);
+        SELECT * FROM mytable;
+        SELECT 1/0;
+        INSERT INTO mytable VALUES(2);"
+    }'
+    ```
+
+    The response would be an array of only one element (the error), akin to:
```

Comment:
This is how PG documents there Simple Query protocol:
https://www.postgresql.org/docs/current/protocol-flow.html#id-1.10.5.7.4

Quoting:

When a simple Query message contains more than one SQL statement (separated by semicolons), those statements are executed as a single transaction, unless explicit transaction control commands are included to force a different behavior. For example, if the message contains

```sql
INSERT INTO mytable VALUES(1);
SELECT 1/0;
INSERT INTO mytable VALUES(2);
```

then the divide-by-zero failure in the SELECT will force rollback of the first INSERT. Furthermore, because execution of the message is abandoned at the first error, the second INSERT is never attempted at all.

end quote

Do we have a custom protocol we're modeling the desired behavior after?

Reply:
Ah, I see what this is saying now--apologies for my misread.

Reply:
This seems like a bad API expecting users to perform their own transaction partitioning, but I digress. I'll amend the design doc.